src/readline.rs

```diff
@@ -51,20 +53,18 @@ impl Highlighter for LineHighlighter {
         let op = eval_math_expression(line, prev_ans);
         match op {
             Ok(_) => {
-                let constants = ["e", "pi"];
-                let functions = [
-                    "sin", "cos", "tan", "csc", "sec", "cot", "sinh", "cosh", "tanh", "ln", "log",
-                    "sqrt", "ceil", "floor", "rad", "deg", "abs", "asin", "acos", "atan", "acsc",
-                    "asec", "acot",
-                ];
+                let constants = CONSTANTS.keys();
+                let functions = FUNCTIONS.keys();
                 let ops = Regex::new(r"(?P<o>[\+-/\*%\^!])").unwrap();
                 let mut coloured: String = ops.replace_all(line, "\x1b[35m$o\x1b[0m").into();
 
-                for c in &constants {
-                    coloured = coloured.replace(c, &format!("\x1b[33m{}\x1b[0m", c));
+                for c in constants {
+                    let re = Regex::new(format!("(?P<o>{})(?P<r>(\x1b\\[35m)?([\\+-/\\*%\\^! ]|$))", c).as_str()).unwrap();
```

Comment:
I think you might want to leave a comment for this regex to let others know what is it doing, it's not short like the normal ones.